### PR TITLE
Add eos_snmp_server resource module

### PR DESCRIPTION
**Pull request by GomathiselviS**
Signed-off-by: GomathiselviS <gomathiselvi@gmail.com>

Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/1236
Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/1237

##### SUMMARY

Add eos_snmp_server resource module.

##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: bindep.txt [hunk 1]
```diff
@@ -4,3 +4,7 @@
 gcc-c++ [doc test platform:rpm]
 python3-devel [test platform:rpm]
 python3 [test platform:rpm]
+libssh-devel [test platform:rpm]
+libffi-devel [test platform:rpm]
+openssl-devel [test platform:rpm]
+libselinux-python [test platform:rpm]
```
Comment (goneri) [inline bindep.txt:10]: please add:

```
libssh-devel [test platform:rpm]
```

**Comment by goneri** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Comment by ashwini-mhatre** [comment]
@GomathiselviS please add new module entry in runtime.yaml

File: plugins/module_utils/network/eos/rm_templates/snmp_server.py [hunk 2]
```diff
@@ -0,0 +1,1227 @@
...
+        command += " sequence-number-skip"
+
+    return command
+
+
+def _tmplt_snmp_server_traps_lldp(config_data):
+    command = "snmp-server enable traps lldp"
+    el = config_data["traps"]["lldp"]
+    if el.get("rem_tables_change"):
+        command += " rem_tables_change"
```
Comment (ashwini-mhatre) [inline plugins/module_utils/network/eos/rm_templates/snmp_server.py:142]: I think this should be rem-tables-change

File: plugins/module_utils/network/eos/rm_templates/snmp_server.py [hunk 3]
```diff
@@ -0,0 +1,1227 @@
...
+            "setval": "snmp-server transmit max-size {{ transmit }}",
+            "result": {
+                "transmit": "{{ num }}"
+            },
+        },
+        {
+            "name": "transport",
+            "getval": re.compile(
+                r"""
+                \s*snmp-server\stransport\stcp
```
Comment (ashwini-mhatre) [inline plugins/module_utils/network/eos/rm_templates/snmp_server.py:1098]: is variable missing? (?P<tcp>) something like this

Comment (GomathiselviS) [reply plugins/module_utils/network/eos/rm_templates/snmp_server.py:1098]: No. As of now only tcp is supported under transport.

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Review by ashwini-mhatre — APPROVED** [review]

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Pull request merged**